test/e2e/kubectl.go

```diff
@@ -190,6 +195,101 @@ var _ = Describe("Kubectl client", func() {
 			}
 		})
 
+		It("support exec through an HTTP proxy", func() {
+			SkipIfProviderIs("local")
+			// Get the access token
+			token := runKubectl("config", "view", "-o", "template", "--template={{range .users}}{{ if eq .name \"e2e\" }}{{ index .user.token }}{{end}}{{end}}")
+			if token == "" {
+				Failf("No token found for e2e user. Output: %s", token)
+			}
+
+			// start exec-proxy-tester container
+			netexecPodPath := filepath.Join(testContext.RepoRoot, "contrib/for-tests/netexec/pod.yaml")
+			runKubectl("create", "-f", netexecPodPath, fmt.Sprintf("--namespace=%v", ns))
+			checkPodsRunningReady(c, ns, []string{netexecContainer}, podStartTimeout)
+
+			pipeReader, postBodyWriter, err := newStreamingUpload(testContext.KubectlPath)
+
+			if err != nil {
+				Failf("unable to create streaming upload. Error: %s", err)
+			}
+
+			By("uploading kubectl to netexec")
+			// Upload the kubectl binary
+			resp, err := c.Post().
+				Prefix("proxy").
+				Namespace(ns).
+				Name("netexec").
+				Resource("pods").
+				Suffix("upload").
+				SetHeader("Content-Type", postBodyWriter.FormDataContentType()).
+				Body(pipeReader).
+				Do().Raw()
+
+			if err != nil {
+				Failf("Unable to upload kubectl binary to the remote exec server due to error: %s", err)
+			}
+
+			uploadBinaryName := string(resp)
+			// Verify that we got the expected response back in the body
+			if !strings.HasPrefix(uploadBinaryName, "/uploads/") {
+				Failf("Unable to upload kubectl binary to remote exec server. /uploads/ not in response. Response: %s", uploadBinaryName)
+			}
+
+			// Fail if the variable isn't set
+			if testContext.Host == "" {
```

Comment:
Move this check up earlier, so we don't even try to start pods, upload, etc. if it's not set. Fail fast! :smile: